### DISCUSSION TITLE
Future releases of Spock

### DISCUSSION THREAD

**Discussion by tenleo** [spockframework/spock#1866] [Q&A] [answered]
Hi,

Is there a roadmap or plan available for Spock, to see what we can expect in terms of Spock releases?

The last milestone release 2.4-M1 dates back to December 2022, and the last official 2.3 release was in September 2022. So there hasn't been a release in well over a year. This had led to concerns within my organization whether Spock is still the right choice for us when writing unit tests. 

Would be great if you can give us an update on what's coming in terms of features and releases. Thanks!

Best regards,
Leon

**Comment by leonard84** [comment, marked as answer]
> So there hasn't been a release in well over a year. This had led to concerns within my organization whether Spock is still the right choice for us when writing unit tests.

Can you expand on that a bit, are you waiting on a specific fix, or is it just a general concern?
I would consider Spock a highly mature framework, sure there is always something that can be tweaked and improved, but you won't get any revolutionary new things (hoping that the JUnit Platform stays around for a long time).

That being said, there are a lot of changes on `master` that warrant a new milestone and after that a GA release. I can't promise a specific date for it though, as this depends on the time and availability of multiple people that are quite busy.

**Reply by tenleo** [reply]
Thanks for the quick reply!

The concerns are that Spock might not be maintained anymore or won't be updated with e.g. support for future Java releases, or Groovy releases. If that should happen, Spock might become unusable at some point. And if we'd had to rewrite all of our tests because they stopped working, that would be a lot of work at once.

So there has been some discussion whether we should slowly migrate to JUnit 5, meaning we'd write new tests in JUnit 5 and migrate tests that we touch when making changes to our applications. But before we make any decisions, we agreed to check in with you guys first. To hopefully get an idea what the future of Spock is.

**Reply by NilsKula** [reply]
Hello, was looking for this as well. For me in particular one needed thing is mocking final classes for example. Feel like im coming across them more and more often.

**Reply by ppazos** [reply]
Same question, how many maintainers does Spock have today? With 200+ open issues I would expect some more frequent releases, even with small improvements. I love Spock and for my use cases it works as expected. The only case that I would add some improvements is in the report customization, and more detailed report. For instance if you check Robot Framework, the report can be traced almost line by line, not only for the asserts. I would love to see a more detailed report also on Spock. I'll be happy to discuss in another thread. All the best from Uruguay.

**Reply by AndreasTu** [reply]
@ppazos I would not count the 200+ issue, if you filter for bugs there are currently only 20.
Many of the issues are nice to have features, old stuff, 3-rd party issue we are still tracking there like Groovy issues etc.

So the simple issue number does not say anything about the state of the project. 
If you look at the commit fequency of master there is quite a lot of stuff happening.

**Reply by leonard84** [reply]
[Spock 2.4-M2](https://github.com/spockframework/spock/releases/tag/spock-2.4-M2) has been released.
